api/src/main/java/io/strimzi/api/kafka/model/KafkaRebalanceSpec.java

```diff
@@ -62,6 +63,17 @@ public void setSkipHardGoalCheck(boolean skipHardGoalCheck) {
         this.skipHardGoalCheck = skipHardGoalCheck;
     }
 
+    @Description("Whether to balance load between disks within brokers, known as intra-broker disk balancing (requires JBOD Kafka deployment. ) " +
+            "Default is false.")
```

Comment:
One more stupid question ... I know is this is a new feature. But is there a reason to disable it by default? It sounds like something useful what would not do any harm if enabled. But maybe I'm missing something,.

Reply:
Never are stupid questions, they are helping me be more thorough! The reason it is disabled by default is because an exception will be thrown if a rebalance proposal is requested with it enabled when JBOD storage is disabled

Reply:
Yeah you are right, there are just a couple of things I need to check in the CC code but I don't think this should have any negative consequences. I'll add something for it in the next commit

Reply:
I would agree with Jakub. Enabled by default and checking if JBOD is used, then at least logging a WARN message to make the user aware that this CC feature won't be used because the Kafka instance is not using JBOD.

Reply:
Good idea guys, I have just added some logic to handle this

Reply:
Alright after much experimentation and investigation of the Cruise Control codebase I have found that inter-broker balancing and intra-broker balancing can only happen as separate requests. This means that JBOD users would need two separate sequential requests to balance between brokers and JBOD disks. For this reason, I am going to revert the `rebalanceDisk` to default to `false` since CC only preforms an intra-broker balance when `rebalanceDisk` is set to `true`!

TL;DR In case I forget, I am going to write down how the goals are chosen when `rebalanceDisks` parameter is set to true. First there is a check to make sure no other goals are specified in the request, if there are CC will throw an error, if not CC will add the intra-broker goals to the goal list [1]. Next, CC will create the `goalsByPriority` list using this goal list [2]. Then the GoalOptimizer will iterate through this `goalsByPriority` list (which now only includes intra-broker goals) to come up with an optimization proposal based on those goals [3]

[1] https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_4/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterUtils.java#L776-L780
[2] https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_4/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlUtils.java#L469-L474
[3] https://github.com/linkedin/cruise-control/blob/migrate_to_kafka_2_4/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java#L443

Reply:
I guess it should be added to the docs directly here since you already modify them?

Reply:
Added some new docs here and in the documentation, going to ping for another review once the tests pass